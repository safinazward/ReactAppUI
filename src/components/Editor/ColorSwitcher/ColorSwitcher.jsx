import React, { Component } from 'react';
import PropTypes from 'prop-types';
import reactCSS from 'reactcss';
import { SketchPicker } from 'react-color';


class ColorSwitcher extends Component {

	constructor(props) {
		super(props);
		this.state = { displayColorPicker: false };
		this.handleClick = this.handleClick.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleChange = this.handleChange.bind(this);
	}

	handleClick = () => {
		this.setState({ displayColorPicker: !this.state.displayColorPicker });
	};

	handleClose = () => {
		this.setState({ displayColorPicker: false });
	};

	handleChange = color => {
		this.props.handleChange(this.props.colorVar, color && color.hex);
	};

	render() {
		const styles = reactCSS({
			default: {
				color: {
					width: '36px',
					height: '14px',
					borderRadius: '2px',
					background: `${this.props.color}`
				},
				swatch: {
					padding: '5px',
					background: '#fff',
					borderRadius: '1px',
					boxShadow: '0 0 0 1px rgba(0,0,0,.1)',
					display: 'inline-block',
					cursor: 'pointer'
				},
				popover: {
					position: 'absolute',
					zIndex: '2'
				},
				cover: {
					position: 'fixed',
					top: '0px',
					right: '0px',
					bottom: '0px',
					left: '0px'
				}
			}
		});

		return (
			<div className="row">
				<label className="col-sm-7 p-2">{this.props.title}</label>
				<div className="col-sm-5 pt-2">
					<div id="showPicker" style={styles.swatch} onClick={this.handleClick}>
						<div style={styles.color} />
					</div>
				</div>
				{this.state.displayColorPicker && (
					<div style={styles.popover}>
						<div id="closePicker" style={styles.cover} onClick={this.handleClose} />
						<SketchPicker
							color={this.props.color}
							onChange={this.handleChange}
						/>
					</div>
				)}
			</div>
		);
	}
}

ColorSwitcher.propTypes= {
	colorVar: PropTypes.string,
	color: PropTypes.string,	
	title: PropTypes.string,
	handleChange: PropTypes.func
}

export default ColorSwitcher;
