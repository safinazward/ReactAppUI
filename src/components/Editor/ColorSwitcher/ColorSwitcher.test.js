import React from 'react';
import { mount, shallow } from 'enzyme';
import { SketchPicker } from 'react-color';

import ColorSwitcher from './ColorSwitcher';

describe('The color switcher', () => {
	it('renders correctly', () => {
		const ColorSwitcherComponent = shallow(< ColorSwitcher />);
		expect(ColorSwitcherComponent).toMatchSnapshot();
	});

	it('should render title', () => {
		const wrapper = mount(< ColorSwitcher title="Primary Color" />);
		expect(wrapper.find('label').text()).toEqual('Primary Color');
	});

	it('should display SketchPicker when clicking on show picker button', () => {
		const wrapper = mount(< ColorSwitcher />);
		const showPickerBtn = wrapper.find('#showPicker');
		showPickerBtn.simulate('click');
		wrapper.update();
		const sketchPicker = wrapper.find('.sketch-picker');
		expect(sketchPicker.length).toEqual(1);
	});

	it('should close SketchPicker when clicking on close picker button', () => {
		const wrapper = mount(< ColorSwitcher />);
		const showPickerBtn = wrapper.find('#showPicker');
		showPickerBtn.simulate('click');
		wrapper.update();
		expect(wrapper.find('.sketch-picker').length).toEqual(1);
		const closePickerBtn = wrapper.find('#closePicker');
		closePickerBtn.simulate('click');
		wrapper.update();
		expect(wrapper.find('.sketch-picker').length).toEqual(0);
	});

	it('should display SketchPicker when clicking on show picker button', () => {
    const handleChange = jest.fn();
    const wrapper = shallow( < ColorSwitcher handleChange = {handleChange} /> );
		wrapper.setState({displayColorPicker:true});
    const showPickerBtn = wrapper.find(SketchPicker);
    showPickerBtn.simulate('change');
    //wrapper.update();
    //const sketchPicker = wrapper.find('.sketch-picker');
    //sketchPicker.children().first().simulate('click');
    expect(handleChange).toHaveBeenCalled();
  });
});