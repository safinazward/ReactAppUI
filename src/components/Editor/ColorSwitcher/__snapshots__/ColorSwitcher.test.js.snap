// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The color switcher renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColorSwitcher />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="col-sm-7 p-2"
/>,
        <div
          className="col-sm-5 pt-2"
>
          <div
                    id="showPicker"
                    onClick={[Function]}
                    style={
                              Object {
                                        "MozBorderRadius": "1px",
                                        "MozBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                        "OBorderRadius": "1px",
                                        "OBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                        "WebkitBorderRadius": "1px",
                                        "WebkitBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                        "background": "#fff",
                                        "borderRadius": "1px",
                                        "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                        "cursor": "pointer",
                                        "display": "inline-block",
                                        "msBorderRadius": "1px",
                                        "msBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                        "padding": "5px",
                                      }
                    }
          >
                    <div
                              style={
                                        Object {
                                                  "MozBorderRadius": "2px",
                                                  "OBorderRadius": "2px",
                                                  "WebkitBorderRadius": "2px",
                                                  "background": "undefined",
                                                  "borderRadius": "2px",
                                                  "height": "14px",
                                                  "msBorderRadius": "2px",
                                                  "width": "36px",
                                                }
                              }
                    />
          </div>
</div>,
        false,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "col-sm-7 p-2",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            id="showPicker"
            onClick={[Function]}
            style={
                        Object {
                                    "MozBorderRadius": "1px",
                                    "MozBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                    "OBorderRadius": "1px",
                                    "OBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                    "WebkitBorderRadius": "1px",
                                    "WebkitBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                    "background": "#fff",
                                    "borderRadius": "1px",
                                    "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                    "cursor": "pointer",
                                    "display": "inline-block",
                                    "msBorderRadius": "1px",
                                    "msBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                    "padding": "5px",
                                  }
            }
>
            <div
                        style={
                                    Object {
                                                "MozBorderRadius": "2px",
                                                "OBorderRadius": "2px",
                                                "WebkitBorderRadius": "2px",
                                                "background": "undefined",
                                                "borderRadius": "2px",
                                                "height": "14px",
                                                "msBorderRadius": "2px",
                                                "width": "36px",
                                              }
                        }
            />
</div>,
          "className": "col-sm-5 pt-2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              style={
                            Object {
                                          "MozBorderRadius": "2px",
                                          "OBorderRadius": "2px",
                                          "WebkitBorderRadius": "2px",
                                          "background": "undefined",
                                          "borderRadius": "2px",
                                          "height": "14px",
                                          "msBorderRadius": "2px",
                                          "width": "36px",
                                        }
              }
/>,
            "id": "showPicker",
            "onClick": [Function],
            "style": Object {
              "MozBorderRadius": "1px",
              "MozBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
              "OBorderRadius": "1px",
              "OBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
              "WebkitBorderRadius": "1px",
              "WebkitBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
              "background": "#fff",
              "borderRadius": "1px",
              "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
              "cursor": "pointer",
              "display": "inline-block",
              "msBorderRadius": "1px",
              "msBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
              "padding": "5px",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "style": Object {
                "MozBorderRadius": "2px",
                "OBorderRadius": "2px",
                "WebkitBorderRadius": "2px",
                "background": "undefined",
                "borderRadius": "2px",
                "height": "14px",
                "msBorderRadius": "2px",
                "width": "36px",
              },
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="col-sm-7 p-2"
/>,
          <div
            className="col-sm-5 pt-2"
>
            <div
                        id="showPicker"
                        onClick={[Function]}
                        style={
                                    Object {
                                                "MozBorderRadius": "1px",
                                                "MozBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                                "OBorderRadius": "1px",
                                                "OBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                                "WebkitBorderRadius": "1px",
                                                "WebkitBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                                "background": "#fff",
                                                "borderRadius": "1px",
                                                "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                                "cursor": "pointer",
                                                "display": "inline-block",
                                                "msBorderRadius": "1px",
                                                "msBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                                "padding": "5px",
                                              }
                        }
            >
                        <div
                                    style={
                                                Object {
                                                            "MozBorderRadius": "2px",
                                                            "OBorderRadius": "2px",
                                                            "WebkitBorderRadius": "2px",
                                                            "background": "undefined",
                                                            "borderRadius": "2px",
                                                            "height": "14px",
                                                            "msBorderRadius": "2px",
                                                            "width": "36px",
                                                          }
                                    }
                        />
            </div>
</div>,
          false,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "col-sm-7 p-2",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              id="showPicker"
              onClick={[Function]}
              style={
                            Object {
                                          "MozBorderRadius": "1px",
                                          "MozBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                          "OBorderRadius": "1px",
                                          "OBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                          "WebkitBorderRadius": "1px",
                                          "WebkitBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                          "background": "#fff",
                                          "borderRadius": "1px",
                                          "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                          "cursor": "pointer",
                                          "display": "inline-block",
                                          "msBorderRadius": "1px",
                                          "msBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                                          "padding": "5px",
                                        }
              }
>
              <div
                            style={
                                          Object {
                                                        "MozBorderRadius": "2px",
                                                        "OBorderRadius": "2px",
                                                        "WebkitBorderRadius": "2px",
                                                        "background": "undefined",
                                                        "borderRadius": "2px",
                                                        "height": "14px",
                                                        "msBorderRadius": "2px",
                                                        "width": "36px",
                                                      }
                            }
              />
</div>,
            "className": "col-sm-5 pt-2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                style={
                                Object {
                                                "MozBorderRadius": "2px",
                                                "OBorderRadius": "2px",
                                                "WebkitBorderRadius": "2px",
                                                "background": "undefined",
                                                "borderRadius": "2px",
                                                "height": "14px",
                                                "msBorderRadius": "2px",
                                                "width": "36px",
                                              }
                }
/>,
              "id": "showPicker",
              "onClick": [Function],
              "style": Object {
                "MozBorderRadius": "1px",
                "MozBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                "OBorderRadius": "1px",
                "OBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                "WebkitBorderRadius": "1px",
                "WebkitBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                "background": "#fff",
                "borderRadius": "1px",
                "boxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                "cursor": "pointer",
                "display": "inline-block",
                "msBorderRadius": "1px",
                "msBoxShadow": "0 0 0 1px rgba(0,0,0,.1)",
                "padding": "5px",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "style": Object {
                  "MozBorderRadius": "2px",
                  "OBorderRadius": "2px",
                  "WebkitBorderRadius": "2px",
                  "background": "undefined",
                  "borderRadius": "2px",
                  "height": "14px",
                  "msBorderRadius": "2px",
                  "width": "36px",
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
